### ISSUE TITLE
3.6 is using vulnerable version of System.Drawing.Common

### ISSUE THREAD

**Issue by flcdrg** [dotnet/Nerdbank.GitVersioning#1089]
According to JFrog XRay, the current 3.6 release is using System.Drawing.Common v4.7.0, which has https://nvd.nist.gov/vuln/detail/CVE-2021-24112

I don't see this as a direct dependency, so assume it must be being pulled in as a transient one.

Upgrading to 4.7.2 or 5.0.3 should apparently resolve this.

**Comment by flcdrg** [comment]
Hmm. so #1090 didn't work. Looking at the assembly that got bundled with 3.6.145 it shows this:
![image](https://github.com/user-attachments/assets/db9e9be7-634a-4c7a-afcb-050ab94a9a6b)

**Comment by flcdrg** [comment]
Hmm.. and yet that is the netstandard2 binary included in System.Drawing.Common v4.7.2

**Comment by AArnott** [comment]
> According to JFrog XRay, the current 3.6 release is using System.Drawing.Common v4.7.0, which has https://nvd.nist.gov/vuln/detail/CVE-2021-24112

I don't see System.Drawing.Common even mentioned in that advisory.

**Comment by flcdrg** [comment]
It is pretty obscure. If you click through to https://msrc.microsoft.com/update-guide/en-US/advisory/CVE-2021-24112, that mentions libgdiplus which ships in that NuGet package.

**Comment by AArnott** [comment]
> that mentions libgdiplus which ships in that NuGet package.

No it doesn't.

<details>

```
tree Q:\.tools\.nuget\packages\system.drawing.common\4.7.2 /f
Folder PATH listing for volume DevDrive
Volume serial number is 9E79-31FC
Q:\.TOOLS\.NUGET\PACKAGES\SYSTEM.DRAWING.COMMON\4.7.2
│   .nupkg.metadata
│   .signature.p7s
│   Icon.png
│   LICENSE.TXT
│   system.drawing.common.4.7.2.nupkg
│   system.drawing.common.4.7.2.nupkg.sha512
│   system.drawing.common.nuspec
│   THIRD-PARTY-NOTICES.TXT
│   useSharedDesignerContext.txt
│   version.txt
│
├───lib
│   ├───MonoAndroid10
│   │       _._
│   │
│   ├───MonoTouch10
│   │       _._
│   │
│   ├───net461
│   │       System.Drawing.Common.dll
│   │
│   ├───netstandard2.0
│   │       System.Drawing.Common.dll
│   │
│   ├───xamarinios10
│   │       _._
│   │
│   ├───xamarinmac20
│   │       _._
│   │
│   ├───xamarintvos10
│   │       _._
│   │
│   └───xamarinwatchos10
│           _._
│
├───ref
│   ├───MonoAndroid10
│   │       _._
│   │
│   ├───MonoTouch10
│   │       _._
│   │
│   ├───net461
│   │       System.Drawing.Common.dll
│   │
│   ├───netcoreapp3.0
│   │       System.Drawing.Common.dll
│   │       System.Drawing.Common.xml
│   │
│   ├───netstandard2.0
│   │       System.Drawing.Common.dll
│   │
│   ├───xamarinios10
│   │       _._
│   │
│   ├───xamarinmac20
│   │       _._
│   │
│   ├───xamarintvos10
│   │       _._
│   │
│   └───xamarinwatchos10
│           _._
│
└───runtimes
    ├───unix
    │   └───lib
    │       ├───netcoreapp2.0
    │       │       System.Drawing.Common.dll
    │       │
    │       └───netcoreapp3.0
    │               System.Drawing.Common.dll
    │               System.Drawing.Common.xml
    │
    └───win
        └───lib
            ├───netcoreapp2.0
            │       System.Drawing.Common.dll
            │
            └───netcoreapp3.0
                    System.Drawing.Common.dll
                    System.Drawing.Common.xml
```

</details>

**Comment by flcdrg** [comment]
My mistake.. I've been looking at too many build outputs and package contents lately and got confused there

**Comment by AArnott** [comment]
So do we have a bug here at all?

**Comment by flcdrg** [comment]
Good question. I might see if I can get some clarification from JFrog to understand why they're flagging System.Drawing.Common and not just libgdiplus

**Comment by flcdrg** [comment]
While searching, I did find https://github.com/advisories/GHSA-rxg9-xrhp-64gj

**Comment by AArnott** [comment]
That one should be addressed by #1090.

**Comment by flcdrg** [comment]
I've done some further testing and I'm pretty sure JFrog Xray is raising a false positive on the System.Drawing.Common.dll. I'm following up with JFrog support, but I think we've done all we need on this side.
